tests/src/Interop/StructMarshalling/PInvoke/MarshalStructAsParamDLL.h

```diff
@@ -374,8 +374,8 @@ void ChangeNumberSequential(NumberSequential* p)
 
 bool IsCorrectNumberSequential(NumberSequential* p)
 {
-	if(p->i32 != (-0x7fffffff - 1) || p->ui32 != 0xffffffff || p->s1 != -0x8000 || p->us1 != 0xffff || p->b != 0 || 
-		p->sb != 0x7f ||p->i16 != -0x8000 || p->ui16 != 0xffff || p->i64 != -1234567890 ||
+	if(p->i32 != (-0x7fffffff - 1) || p->ui32 != 0xffffffff || p->s1 != -0x8000 || p->us1 != 0xffff || p->b != 0 ||
+		p->sb != 0x7f ||p->i16 != -0x8000 || p->ui16 != 0xffff || p->i64 != (uint64_t)-1234567890 ||
```

Comment:
This doesn't look right, the p->i64 is signed and you are now comparing it to 32 bit unsigned value. It seems this cast should not be needed at all, the `-1234567890` fits into 32 bit signed int and so comparing it with 64 bit signed int should not generate any warnings.

Reply:
Just changed it to `uint64_t` some time ago to see if `ReversePInvoke_MaralSeqStruct` test passes.
(on a different machine now, I will post the gcc warning shortly.)

Reply:
gcc erros out like this:

```sh
In file included from /coreclr/tests/src/Interop/StructMarshalling/PInvoke/MarshalStructAsParamDLL.cpp:1:0:
/coreclr/tests/src/Interop/StructMarshalling/PInvoke/MarshalStructAsParamDLL.h: In function 'bool IsCorrectNumberSequential(NumberSequential*)':
/coreclr/tests/src/Interop/StructMarshalling/PInvoke/MarshalStructAsParamDLL.h:378:68: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
   p->sb != 0x7f ||p->i16 != -0x8000 || p->ui16 != 0xffff || p->i64 != -1234567890 ||
                                                             ~~~~~~~^~~~~~~~~~~~~~
```

with `uint64_t` cast, both clang and gcc are satisfied.

Reply:
I see - there is a bug in the LONG64 definition. It should be defined as "long long", not "unsigned long long". That's causing the issue.
It is defined in tests\src\Interop\common\xplatform.h.

Reply:
Thanks, fixed LONG64 definition and reverted the cast.